### ISSUE TITLE
Zed keeps getting stuck "resolving code actions"

### ISSUE THREAD

**Issue by dela-dels** [zed-industries/zed#20249]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

While using Zed, i've noticed for some time now that the editor gets stuck "Resolving code actions". This i've noticed prevents intellisense from working properly as well. I'm mostly writing PHP code.

### Environment

Zed: v0.159.10 (Zed)
OS: macOS 15.1.0
Memory: 8 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

![Screenshot 2024-11-05 at 6 02 22 PM](https://github.com/user-attachments/assets/e159d4aa-5147-454c-872c-9d08924e914f)

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

2024-11-05T17:58:37.580406Z [ERROR] sending into a closed channel
2024-11-05T17:58:37.581326Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:37.663478Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:37.663716Z [ERROR] server shut down
2024-11-05T17:58:37.747575Z [ERROR] sending into a closed channel
2024-11-05T17:58:37.748449Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:37.797017Z [ERROR] sending into a closed channel
2024-11-05T17:58:37.797915Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:37.881485Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:37.881721Z [ERROR] server shut down
2024-11-05T17:58:37.984785Z [ERROR] sending into a closed channel
2024-11-05T17:58:37.985912Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:38.066931Z [ERROR] sending into a closed channel
2024-11-05T17:58:38.068146Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:38.116598Z [ERROR] sending into a closed channel
2024-11-05T17:58:38.118154Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:38.19809Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:38.198326Z [ERROR] server shut down
2024-11-05T17:58:38.299521Z [ERROR] sending into a closed channel
2024-11-05T17:58:38.300759Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:38.367097Z [ERROR] sending into a closed channel
2024-11-05T17:58:38.368297Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:38.450416Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:38.450742Z [ERROR] server shut down
2024-11-05T17:58:38.481623Z [ERROR] sending into a closed channel
2024-11-05T17:58:38.482769Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:38.565248Z [ERROR] sending into a closed channel
2024-11-05T17:58:38.566365Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:38.647582Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:38.647819Z [ERROR] server shut down
2024-11-05T17:58:38.797787Z [ERROR] sending into a closed channel
2024-11-05T17:58:38.798838Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:38.870028Z [ERROR] sending into a closed channel
2024-11-05T17:58:38.879344Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:38.950015Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:38.950385Z [ERROR] server shut down
2024-11-05T17:58:38.986608Z [ERROR] sending into a closed channel
2024-11-05T17:58:38.997974Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:39.06641Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:39.066773Z [ERROR] server shut down
2024-11-05T17:58:39.084309Z [ERROR] sending into a closed channel
2024-11-05T17:58:39.085593Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:39.162385Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:39.162609Z [ERROR] server shut down
2024-11-05T17:58:39.265984Z [ERROR] sending into a closed channel
2024-11-05T17:58:39.267212Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:39.347765Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:39.348003Z [ERROR] server shut down
2024-11-05T17:58:39.533326Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:39.533646Z [ERROR] server shut down
2024-11-05T17:58:39.56567Z [ERROR] sending into a closed channel
2024-11-05T17:58:39.566914Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:39.646111Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:39.646346Z [ERROR] server shut down
2024-11-05T17:58:39.686944Z [ERROR] sending into a closed channel
2024-11-05T17:58:39.696345Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:39.781507Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:39.781825Z [ERROR] server shut down
2024-11-05T17:58:39.846782Z [ERROR] sending into a closed channel
2024-11-05T17:58:39.847844Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:39.931915Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:39.932154Z [ERROR] server shut down
2024-11-05T17:58:40.080451Z [ERROR] sending into a closed channel
2024-11-05T17:58:40.081612Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:40.163853Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:40.164103Z [ERROR] server shut down
2024-11-05T17:58:40.252566Z [ERROR] sending into a closed channel
2024-11-05T17:58:40.262038Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:40.33145Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:40.331686Z [ERROR] server shut down
2024-11-05T17:58:40.532533Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:40.532812Z [ERROR] server shut down
2024-11-05T17:58:40.963096Z [ERROR] sending into a closed channel
2024-11-05T17:58:40.966666Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:40.967681Z [ERROR] server shut down
2024-11-05T17:58:40.967875Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:41.045014Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:41.045337Z [ERROR] server shut down
2024-11-05T17:58:41.195018Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:41.195276Z [ERROR] server shut down
2024-11-05T17:58:41.378409Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:41.378621Z [ERROR] server shut down
2024-11-05T17:58:42.598382Z [ERROR] sending into a closed channel
2024-11-05T17:58:42.599667Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:42.599894Z [ERROR] server shut down
2024-11-05T17:58:42.681296Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:42.681533Z [ERROR] server shut down
2024-11-05T17:58:42.862609Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:42.862817Z [ERROR] server shut down
2024-11-05T17:58:43.116293Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:43.382311Z [ERROR] sending into a closed channel
2024-11-05T17:58:43.415226Z [ERROR] sending into a closed channel
2024-11-05T17:58:43.647579Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:43.647795Z [ERROR] server shut down
2024-11-05T17:58:43.966248Z [ERROR] sending into a closed channel
2024-11-05T17:58:48.012357Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:48.012609Z [ERROR] server shut down
2024-11-05T17:58:48.147666Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:48.147909Z [ERROR] server shut down
2024-11-05T17:58:48.332729Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:48.333067Z [ERROR] server shut down
2024-11-05T17:58:48.631514Z [ERROR] sending into a closed channel
2024-11-05T17:58:48.632978Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:48.713877Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:48.714118Z [ERROR] server shut down
2024-11-05T17:58:48.765764Z [ERROR] sending into a closed channel
2024-11-05T17:58:48.766936Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:48.847555Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:48.847806Z [ERROR] server shut down
2024-11-05T17:58:48.968962Z [ERROR] sending into a closed channel
2024-11-05T17:58:48.978584Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:49.036328Z [ERROR] sending into a closed channel
2024-11-05T17:58:49.04642Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:49.132869Z [ERROR] sending into a closed channel
2024-11-05T17:58:49.13445Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:49.214025Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:49.214258Z [ERROR] server shut down
2024-11-05T17:58:49.253632Z [ERROR] sending into a closed channel
2024-11-05T17:58:49.263197Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:49.348874Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:49.349231Z [ERROR] server shut down
2024-11-05T17:58:49.532105Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T17:58:49.532384Z [ERROR] server shut down
2024-11-05T18:01:28.548736Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T18:01:28.549918Z [ERROR] server shut down
2024-11-05T18:01:28.648162Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T18:01:28.648362Z [ERROR] server shut down
2024-11-05T18:01:28.831614Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
2024-11-05T18:01:28.831997Z [ERROR] server shut down

```
<!-- Click above this line and paste or drag-and-drop your log--></details>

**Comment by SomeoneToIgnore** [comment]
This is due to the php language server (it is a good idea to specify your language in the issue too) failing.

>  2024-11-05T17:58:38.300759Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down

Do you have the `phpactor` logs somewhere? 
Otherwise it's not very clear how to help you.

**Comment by SomeoneToIgnore** [comment]
Alternatively, please share the project we can repro the issue on.

**Comment by dela-dels** [comment]
> This is due to the php language server (it is a good idea to specify your language in the issue too) failing.
> 
> > 2024-11-05T17:58:38.300759Z [WARN] Generic lsp request to phpactor/phpactor failed: server shut down
> 
> Do you have the `phpactor` logs somewhere? Otherwise it's not very clear how to help you.

@SomeoneToIgnore unfortunately I've not been successful getting the logs of phpfactor

**Comment by dela-dels** [comment]
> Alternatively, please share the project we can repro the issue on.

https://github.com/dela-dels/zed-phpfactor-issue

sample repo. can't share the actual project cos company rules

**Comment by SomeoneToIgnore** [comment]
Thank you, I have been testing this but unable to see anything odd with the LSP, it definitely does not crash for me:

https://github.com/user-attachments/assets/99da2b76-ae23-4530-a279-1a7245947dc4

The problem is — I do not see any code actions at all there, so presumably it's due to some extra phpfactor configuration?
I have not written any php code ever and did not configure the LSP server anyhow specifically.

**Comment by dela-dels** [comment]
> Thank you, I have been testing this but unable to see anything odd with the LSP, it definitely does not crash for me:
> 
>  phpfactor.mov 
> The problem is — I do not see any code actions at all there, so presumably it's due to some extra phpfactor configuration? I have not written any php code ever and did not configure the LSP server anyhow specifically.

So I noticed as well that with the repo I shared, I do not see those issues. My assumption is that, I face the said issue in the current codebase probably because it's a very huge codebase

**Comment by dela-dels** [comment]
@SomeoneToIgnore managed to get a recording of the issue while working on a codebase today.

https://github.com/user-attachments/assets/a0f6cb2f-0c94-415c-b741-0a4101bcc262

**Comment by SomeoneToIgnore** [comment]
Yes, I'm afraid it's something relatively internal to phpfactor, as Zed does not even have this `Resolving code actions` message, it comes out of the LSP server.

So, we most probably won't know what happens just by looking at Zed, without the server logs or a way to repro this.

**Comment by dela-dels** [comment]
@SomeoneToIgnore Oh right, i get you.

**Comment by aaronliu0130** [comment]
@SomeoneToIgnore Not sure if I should file a new issue instead, but here's my log file, with a logging level of "info" since "debug" went past megabytes in the blink of an eye:

[.phpactor.log](https://github.com/user-attachments/files/18538379/default.phpactor.log)

The LSP stopped working sometime around these lines:

```log
[2025-01-24 15:40:07] phpactor.ERROR: Error when handling "Phpactor\LanguageServer\Core\Rpc\RequestMessage" ({"jsonrpc":"2.0","id":79,"method":"textDocument\/definition","params":{"textDocument":{"uri":"file:\/\/\/home\/aliu\/Documents\/mediawiki\/includes\/skins\/Skin.php"},"position":{"line":1665,"character":10}}}): Exception [Phpactor\TextDocument\Exception\TextDocumentNotFound] Text document "file:///home/aliu/.local/share/zed/extensions/work/php/phpactor-2024.11.28.1/phpactor.phar/vendor/jetbrains/phpstorm-stubs/pcre/pcre.php" not found {"channel":"LSP"} []
[2025-01-24 15:40:19] phpactor.WARNING: The process was killed {"channel":"LSP"} []
[2025-01-24 15:40:19] phpactor.WARNING: Killed process "91437" ('/usr/bin/php' '/home/aliu/.local/share/zed/extensions/work/php/phpactor-2024.11.28.1/phpactor.phar' 'language-server:diagnostics' '--uri=file:///home/aliu/Documents/mediawiki/includes/skins/Skin.php' '--config-extra={"worse_reflection.enable_context_location": false}') because it lived longer than 5s {"channel":"LSP"} []
```

**Comment by SomeoneToIgnore** [comment]
Great, thank you for this.
Not sure it's more clear for me, but at least it's something: the extension seems to lack (?) certain files that phpfactor wants to have?
Those seem to be merely stubs though, so in the worst case, we can add them; but the path with `jetbrains` is somewhat confusing, as if the extension is doing something totally wrong on its side...

Last time, I did not manage to reproduce this error on the public repo given, so this is also not very clear — if you could provide some public example that reliably fails for you and more details if extra set-up is needed (does phpfactor want some config(s) and subscription(s)?) it would be splendid.

**Comment by aaronliu0130** [comment]
I was coding on the MediaWiki-core codebase, specifically includes/skin/skin.php. The extension failed with the default configuration after a seemingly random amount (but less than 10 minutes) time of coding things. you could maybe try coding fizz buzz or something in the aforementioned file to reproduce it? idk

**Comment by mehmetozguldev** [comment]
Any updates?

**Comment by dtannen** [comment]
if this happens to you then just add: 
  "languages": {
    "PHP": {
      "language_servers": ["intelephense", "!phpactor", "..."]
    }
  }
to your settings.

**Comment by aaronliu0130** [comment]
That would disable virtually all PHP linting and the auto-complete and the hover-to-see-symbol-details thing.

**Comment by bdlowery** [comment]
Yes, I have this same problem using Laravel with phpactor. I would save a file 6 times and I'd end up with 6 "resolving code actions" for multiple minutes. I had to switch back to intelephense to get that to stop.

I opened the same project in neovim and saved with phpactor and everything was instant - so it sounds like what @dantleech is saying is correct.

**Comment by dantleech** [comment]
fwiw this has been raised elsewhere, I've created an issue to look into it: https://github.com/phpactor/phpactor/issues/3042

**Comment by shawnwall** [comment]
just noting that php by default in zed is fairly unusable due to the default of `phpactor`. they may be looking into the issue, but just calling it out in the zed repo as it make zed look bad when in fact its default lang server for php that is the problem right now.

**Comment by dela-dels** [comment]
> just noting that php by default in zed is fairly unusable due to the default of `phpactor`. they may be looking into the issue, but just calling it out in the zed repo as it make zed look bad when in fact its default lang server for php that is the problem right now.

Agreed, for what's worth, you can switch to using intelephense or mago. I'm currently using mago which has been pretty fast

**Comment by shawnwall** [comment]
> > just noting that php by default in zed is fairly unusable due to the default of `phpactor`. they may be looking into the issue, but just calling it out in the zed repo as it make zed look bad when in fact its default lang server for php that is the problem right now.
> 
> Agreed, for what's worth, you can switch to using intelephense or mago. I'm currently using mago which has been pretty fast

yup, totally, i've been using intelephense here. i should have called that out, so thanks!

**Comment by dantleech** [comment]
fwiw I'm going to try and look at this in Phpactor today ... 👀

**Comment by dantleech** [comment]
If anybody wants to test (not sure how, maybe by compiling the PHAR) - I've created this PR: it should both limit the "code-action" resolving concurrency to 1 and do it in a separate process to not be blocking: https://github.com/phpactor/phpactor/pull/3048

**Comment by dantleech** [comment]
that version ☝️ has been released and should fix this particular problem.

**Comment by tinhtran24** [comment]
Hi @dantleech i got the warning when open php file on version 
Zed 1.5.3 
Commit: 5d370d67c6ed5a81563695d364385568f6c94be1
Version: 1.5.3+stable.299.5d370d67c6ed5a81563695d364385568f6c94be1
 
<img width="463" height="196" alt="Image" src="https://github.com/user-attachments/assets/fcd85ddf-21c9-47a9-8cf9-50193ff75f99" />

**Comment by dantleech** [comment]
There was an issue resulting in that warning in the Phactor [2026.05.30.0](https://github.com/phpactor/phpactor/releases/tag/2026.05.30.0) release, which was fixed in [2026.05.30.1](https://github.com/phpactor/phpactor/releases/tag/2026.05.30.1). Can you confirm the version of Phactor you are running and if the new version fixes it?

**Comment by tinhtran24** [comment]
Hi @dantleech 
I'm not sure if this was already happening in the previous release, since I haven't been coding in PHP recently. But in the current version I'm running, the warning/error does appear. Here is the full Zed log for reference.

```
2026-06-04T13:09:26+07:00 ERROR [project::lsp_store] Error handling response for request GetCodeActions { range: Anchor { timestamp: Lamport {<local>: 1}, offset: 177, bias: Left, buffer_id: BufferId(12884903190) }..Anchor { timestamp: Lamport {<local>: 1}, offset: 177, bias: Left, buffer_id: BufferId(12884903190) }, kinds: None }: Get code actions via phpactor failed: Exception [Amp\ByteStream\ClosedException] Stream has already been closed. at phar:///Users/tinhtran/Library/Application Support/Zed/extensions/work/php/phpactor-2026.05.30.1/phpactor.phar/vendor/amphp/process/lib/ProcessOutputStream.php#71
2026-06-04T13:10:17+07:00 ERROR [gpui::app] timed out waiting on app_will_quit
2026-06-04T13:10:18+07:00 INFO  [zed] ========== starting zed version 1.5.3+stable.299.5d370d67c6ed5a81563695d364385568f6c94be1, sha 5d370d6 ==========
2026-06-04T13:10:19+07:00 INFO  [crashes] spawning crash handler process
2026-06-04T13:10:19+07:00 INFO  [zed] Using git binary path: "/Applications/Zed.app/Contents/MacOS/git"
2026-06-04T13:10:19+07:00 INFO  [crashes] connected to crash handler process after 100ms
2026-06-04T13:10:19+07:00 INFO  [crashes] panic handler registered
2026-06-04T13:10:19+07:00 INFO  [crashes] crash signal handlers installed
2026-06-04T13:10:19+07:00 INFO  [crashes] crash handler registered
2026-06-04T13:10:19+07:00 INFO  [extension_host] extensions updated. loading 14, reloading 0, unloading 0
2026-06-04T13:10:19+07:00 INFO  [util] set environment variables from shell:/bin/zsh, path:/Users/tinhtran/.bun/bin:/usr/local/opt/icu4c@77/sbin:/usr/local/opt/icu4c@77/bin:/opt/homebrew/Cellar/libpq/15.1/bin:/Users/tinhtran/.yarn/bin:/Users/tinhtran/.config/yarn/global/node_modules/.bin:/usr/local/Caskroom/gcloud-cli/latest/google-cloud-sdk/bin:/usr/local/Caskroom/gcloud-cli/latest/google-cloud-sdk/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/Apple/usr/bin:/usr/local/bin
2026-06-04T13:10:19+07:00 INFO  [auto_update] Auto Update: checking for updates

2026-06-04T13:10:33+07:00 ERROR [project::lsp_store] Error handling response for request GetCodeActions { range: Anchor::min(BufferId(4294967620))..Anchor::min(BufferId(4294967620)), kinds: None }: Get code actions via phpactor failed: Exception [Amp\ByteStream\ClosedException] Stream has already been closed. at phar:///Users/tinhtran/Library/Application Support/Zed/extensions/work/php/phpactor-2026.05.30.1/phpactor.phar/vendor/amphp/process/lib/ProcessOutputStream.php#71
2026-06-04T13:14:31+07:00 ERROR [project::lsp_store] Error handling response for request GetCodeActions { range: Anchor { timestamp: Lamport {<local>: 1}, offset: 294, bias: Left, buffer_id: BufferId(4294967620) }..Anchor { timestamp: Lamport {<local>: 1}, offset: 294, bias: Left, buffer_id: BufferId(4294967620) }, kinds: None }: Get code actions via phpactor failed: Exception [Amp\ByteStream\ClosedException] Stream has already been closed. at phar:///Users/tinhtran/Library/Application Support/Zed/extensions/work/php/phpactor-2026.05.30.1/phpactor.phar/vendor/amphp/process/lib/ProcessOutputStream.php#71
2026-06-04T13:14:33+07:00 ERROR [project::lsp_store] Error handling response for request GetCodeActions { range: Anchor { timestamp: Lamport {<local>: 1}, offset: 37275, bias: Left, buffer_id: BufferId(4294967621) }..Anchor { timestamp: Lamport {<local>: 1}, offset: 37275, bias: Left, buffer_id: BufferId(4294967621) }, kinds: None }: Get code actions via phpactor failed: Exception [Amp\ByteStream\ClosedException] Stream has already been closed. at phar:///Users/tinhtran/Library/Application Support/Zed/extensions/work/php/phpactor-2026.05.30.1/phpactor.phar/vendor/amphp/process/lib/ProcessOutputStream.php#71
2026-06-04T13:14:34+07:00 ERROR [project::lsp_store] Error handling response for request GetCodeActions { range: Anchor { timestamp: Lamport {<local>: 1}, offset: 294, bias: Left, buffer_id: BufferId(4294967620) }..Anchor { timestamp: Lamport {<local>: 1}, offset: 294, bias: Left, buffer_id: BufferId(4294967620) }, kinds: None }: Get code actions via phpactor failed: Exception [Amp\ByteStream\ClosedException] Stream has already been closed. at phar:///Users/tinhtran/Library/Application Support/Zed/extensions/work/php/phpactor-2026.05.30.1/phpactor.phar/vendor/amphp/process/lib/ProcessOutputStream.php#71
2026-06-04T13:14:35+07:00 ERROR [project::lsp_store] Error handling response for request GetCodeActions { range: Anchor { timestamp: Lamport {<local>: 1}, offset: 37275, bias: Left, buffer_id: BufferId(4294967621) }..Anchor { timestamp: Lamport {<local>: 1}, offset: 37275, bias: Left, buffer_id: BufferId(4294967621) }, kinds: None }: Get code actions via phpactor failed: Exception [Amp\ByteStream\ClosedException] Stream has already been closed. at phar:///Users/tinhtran/Library/Application Support/Zed/extensions/work/php/phpactor-2026.05.30.1/phpactor.phar/vendor/amphp/process/lib/ProcessOutputStream.php#71
2026-06-04T13:14:51+07:00 ERROR [crates/install_cli/src/install_cli_binary.rs:31] No such file or directory (os error 2)
```

**Comment by dantleech** [comment]
looks like you have the latest version. That warning is new and can genuinely happen if the file you're editing is "complex" file (1000+ lines of nested if statements for example). Does it happen all the time or just on that file? Are there any other side-effects?

**Comment by dahaupt** [comment]
@dantleech I'm experiencing the same warning. It seems to be caused by code action requests stacking up — every cursor move or click triggers a new "resolving code actions" request before the previous one has finished.

Zed version: 1.6.3
PHPActor version: 2026.05.30.1 (verified with folder `~/.local/share/zed/extensions/work/php/phpactor-2026.05.30.1`)

https://github.com/user-attachments/assets/4e0622ab-af74-4137-b376-87c897b47645

**Comment by dantleech** [comment]
hi, so I don't get this even with a complicated source file (and your rector config doesn't look complicated), the code actions resolve as expected:

<img width="510" height="154" alt="Image" src="https://github.com/user-attachments/assets/8afde7f5-d685-4cd0-b501-0f849d2759cc" />

do you think it would be possible to provide a reproduction repository?

**Comment by dantleech** [comment]
ah, ok, I _do_ get the issue with a larger file. The problem is that it's _cancelling_ the process before starting a new one, but that triggers this warning errorneously. I'll create a fix...

**Comment by dantleech** [comment]
this should fix it: https://github.com/phpactor/phpactor/releases/tag/2026.05.30.2
